tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/annotation/MetaAttributeGenerationVisitor.java

```diff
@@ -138,14 +138,20 @@ private AnnotationMetaAttribute createMetaCollectionAttribute(DeclaredType decla
 				accessTypeInfo.setDefaultAccessType( entity.getEntityAccessTypeInfo().getAccessType() );
 			}
 		}
-		if ( collection.equals( "javax.persistence.metamodel.MapAttribute" ) ) {
+		if ( collection.equals( Constants.MAP_ATTRIBUTE ) ) {
```

Comment:
In the _else_ branch, we handle the basic collection situation when an `AttributeConverter` is applied; however they can also be applied to a `Map` type collection.  A simple example might be something like this:
````
@Convert(converter = KeyValuePairAsJsonSimpleString.class)
private Map<String, Object> keyValuePairs;
````
I'd strongly suggest if we're going to address non-map collections, we might as well address it for these too just to be consistent in the generation of the static metamodel classes.

Reply:
Thanks @Naros. I'll have to investigate that too.